coral/src/app/features/connectors/request/ConnectorEditRequest.test.tsx

```diff
@@ -0,0 +1,453 @@
+      mockGetConnectorDetailsPerEnv.mockResolvedValue(
+        testConnectorDetailsPerEnvResponse
+      );
+      customRender(
+        <Routes>
+          <Route
+            path="/connector/:connectorName/request-update"
+            element={
+              <AquariumContext>
+                <ConnectorEditRequest />
+              </AquariumContext>
+            }
+          />
+        </Routes>,
+        {
+          queryClient: true,
+          memoryRouter: true,
+          customRoutePath: `/connector/${CONNECTOR_NAME}/request-update?env=${ENV_ID}`,
+        }
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
+      cleanup();
+    });
+
+    describe("Renders disabled fields with correct values", () => {
+      it("shows a disabled required select element for 'Environment' with correct value", async () => {
+        const select = await screen.findByRole("combobox", {
+          name: "Environment *",
+        });
+        expect(select).toBeDisabled();
+        expect(select).toBeRequired();
+        expect(select).toHaveDisplayValue(
+          testConnectorDetailsPerEnvResponse.connectorContents.environmentName
+        );
+      });
+
+      it("shows a readOnly required input element for 'Connector name' with correct value", async () => {
+        const input = await screen.findByRole("textbox", {
+          name: "Connector name *",
+        });
+        expect(input).toHaveAttribute("readonly");
+        expect(input).toBeRequired();
+        expect(input).toHaveDisplayValue(CONNECTOR_NAME);
+      });
+    });
+
+    describe("Renders enabled field with correct values", () => {
+      it("should render 'Connector configuration' with correct default value", async () => {
+        const mockedAdvancedConfig = screen.getByTestId(
+          "connector-request-config"
+        );
+
+        expect(mockedAdvancedConfig).toBeEnabled();
+        await waitFor(() =>
+          expect(mockedAdvancedConfig).toHaveDisplayValue(
+            testConnectorDetailsPerEnvResponse.connectorContents.connectorConfig
+          )
+        );
+      });
+    });
+
+    it("should render 'Description' with correct default value", async () => {
+      const connectorNameInput = screen.getByRole("textbox", {
+        name: "Connector description *",
+      });
+      await waitFor(() =>
+        expect(connectorNameInput).toHaveDisplayValue(
+          testConnectorDetailsPerEnvResponse.connectorContents.description
+        )
+      );
+    });
+
+    it("should render 'Message for approval' with no default value", async () => {
+      const remarksInput = screen.getByRole("textbox", {
+        name: "Message for approval",
+      });
+      await user.clear(remarksInput);
+      await user.type(remarksInput, "test");
+      expect(remarksInput).toHaveDisplayValue("test");
+    });
+
+    it("shows a notification and does not submit when user has not changed topic data", async () => {
+      await userEvent.click(
```

Comment:
It's a bit of a nitpick, but I think the pattern:

```
const button = screen.getByRole("button", { name: "Submit update request" })
await userEvent.click(button) 
```

is better to read then having the query inline. Especially in tests later where there are multiple userEvents, it's getting harder to follow the flow.

Reply:
[df8e9ec](https://github.com/aiven/klaw/pull/1523/commits/df8e9ec12184aea733865981eafb4a48eae15f86)